srml/randomness/src/lib.rs

```diff
@@ -0,0 +1,304 @@
+		type Lookup = IdentityLookup<Self::AccountId>;
+		type Header = Header;
+		type WeightMultiplierUpdate = ();
+		type Event = ();
+		type BlockHashCount = BlockHashCount;
+		type MaximumBlockWeight = MaximumBlockWeight;
+		type AvailableBlockRatio = AvailableBlockRatio;
+		type MaximumBlockLength = MaximumBlockLength;
+		type Version = ();
+	}
+
+	type System = system::Module<Test>;
+	type Randomness = Module<Test>;
+
+	fn new_test_ext() -> runtime_io::TestExternalities<Blake2Hasher> {
+		let t = system::GenesisConfig::default().build_storage::<Test>().unwrap();
+		t.into()
+	}
+
+	#[test]
+	fn test_block_number_to_index() {
+		for i in 2 .. 1000 {
+			assert_eq!((i - 2) as usize % 81, block_number_to_index::<Test>(i));
+		}
+	}
+
+	#[test]
+	fn test_random_material_parital() {
+		with_externalities(&mut new_test_ext(), || {
+			let genesis_hash = System::parent_hash();
+			let mut parent_hash = genesis_hash;
+
+			for i in 2..40 {
+				System::initialize(&i, &parent_hash, &Default::default(), &Default::default());
+				Randomness::on_initialize(i);
+
+				let header = System::finalize();
+				parent_hash = header.hash();
+			}
+
+			let random_material = Randomness::random_material();
+
+			assert_eq!(random_material.len(), 38);
+			assert_eq!(random_material[0], genesis_hash);
+		});
+	}
+
+	#[test]
+	fn test_random_material_filled() {
+		with_externalities(&mut new_test_ext(), || {
+			let genesis_hash = System::parent_hash();
+			let mut parent_hash = genesis_hash;
+
+			for i in 2..83 {
+				System::initialize(&i, &parent_hash, &Default::default(), &Default::default());
+				Randomness::on_initialize(i);
+
+				let header = System::finalize();
+				parent_hash = header.hash();
+			}
+
+			let random_material = Randomness::random_material();
+
+			assert_eq!(random_material.len(), 81);
+			assert_ne!(random_material[0], random_material[1]);
+			assert_eq!(random_material[0], genesis_hash);
+		});
+	}
+
+	#[test]
+	fn test_random_material_filled_twice() {
+		with_externalities(&mut new_test_ext(), || {
+			let genesis_hash = System::parent_hash();
+			let mut parent_hash = genesis_hash;
+
+			for i in 2..164 {
+				System::initialize(&i, &parent_hash, &Default::default(), &Default::default());
+				Randomness::on_initialize(i);
+
+				let header = System::finalize();
+				parent_hash = header.hash();
+			}
+
+			let random_material = Randomness::random_material();
+
+			assert_eq!(random_material.len(), 81);
+			assert_ne!(random_material[0], random_material[1]);
+			assert_ne!(random_material[0], genesis_hash);
+		});
+	}
+
+	#[test]
+	fn test_random() {
+		with_externalities(&mut new_test_ext(), || {
+			let mut parent_hash = System::parent_hash();
+
+			for i in 2..164 {
+				System::initialize(&i, &parent_hash, &Default::default(), &Default::default());
+				Randomness::on_initialize(i);
+
+				let header = System::finalize();
+				parent_hash = header.hash();
+				// Why is this needed?
+				System::set_block_number(*header.number());
```

Comment:
because the last thing that you call is `finalize()` which `kill`s `system::Number<T>`